tools/pkglint/lib/rules.ts

```diff
@@ -46,9 +46,32 @@ export class DescriptionIsRequired extends ValidationRule {
   }
 }
 
+/**
+ * Verify that all packages have a publishConfig with a publish tag set.
+ */
+export class PublishConfigTagIsRequired extends ValidationRule {
+  public readonly name = 'package-info/publish-config-tag';
+
+  public validate(pkg: PackageJson): void {
```

Comment:
We can skip this for private packages. This will avoid the changes to `tools/` and others that don't need this section.

```ts
if (pkg.json.private) {  return; }
```

This would also mean that this section need not change for a huge portion of the packages on v2 branch, because it's irrelevant.